src/act.movement.cpp

```diff
@@ -72,7 +72,7 @@ int can_move(struct char_data *ch, int dir, int extra)
   if (IS_SET(extra, CHECK_SPECIAL) && special(ch, convert_dir[dir], &empty_argument))
     return 0;
 
-  if (ch->in_room && ch->in_room->icesheet[0] && !IS_ASTRAL(ch) && !IS_PROJECT(ch)) {
+  if (ch->in_room && ch->in_room->icesheet[0] && !IS_ASTRAL(ch) && !IS_PROJECT(ch) && !IS_AFFECTED(ch, AFF_LEVITATE)) {
```

Comment:
You've used AFF_LEVITATE a lot, but I don't think it's been defined. It goes in awake.h with the other affs, and ~~you'll also need to add entries to constants.cpp to describe it when someone is statted etc~~ looks like you already did that part.